### ISSUE TITLE
BUG: Default Plugins Exception After Updating

### ISSUE THREAD

**Issue by mossadforsale** [Flow-Launcher/Flow.Launcher#2891]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I am using the latest version of Flow Launcher.

### Problem Description

Hello, today I tried to update a bunch of plugins for Flow Launcher by `pm update` command and I've got an error of Explorer plugin:
```
Everything Search Exception:
 Flow.Launcher.Plugin.Explorer.Exceptions.SearchException: Method not found: 'Void PreviewInfo.set_FilePath(System.String)'.
 ---> System.MissingMethodException: Method not found: 'Void PreviewInfo.set_FilePath(System.String)'.
   at Flow.Launcher.Plugin.Explorer.Search.ResultManager.CreateFolderResult(String title, String subtitle, String path, Query query, Int32 score, Boolean windowsIndexed)
   at Flow.Launcher.Plugin.Explorer.Search.ResultManager.CreateResult(Query query, SearchResult result) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\ResultManager.cs:line 67
   at Flow.Launcher.Plugin.Explorer.Search.SearchManager.SearchAsync(Query query, CancellationToken token) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\SearchManager.cs:line 116
   at Flow.Launcher.Plugin.Explorer.Search.SearchManager.SearchAsync(Query query, CancellationToken token) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\SearchManager.cs:line 112
   --- End of inner exception stack trace ---
   at Flow.Launcher.Plugin.Explorer.Search.SearchManager.SearchAsync(Query query, CancellationToken token) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\SearchManager.cs:line 129
   at Flow.Launcher.Plugin.Explorer.Main.QueryAsync(Query query, CancellationToken token) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.Explorer\Main.cs:line 62
```
When I tried to reinstall Explorer plugin by Plugin Manager, I've got the exception message:
```
Flow Launcher version: 1.18.0
OS Version: 22631.3958
IntPtr Length: 8
x64: True

Python Path: C:\Users\mossadforsale\software\FlowLauncher\app-1.18.0\UserData\Environments\Python\PythonEmbeddable-v3.11.4\pythonw.exe
Node Path: C:\Users\mossadforsale\software\FlowLauncher\app-1.18.0\UserData\Environments\Node.js\Node-v16.18.0\node-v16.18.0-win-x64\node.exe
Date: 08/14/2024 12:33:09
Exception:
Plugins Manager Exception: 
Websites: https://github.com/Flow-Launcher/Flow.Launcher
Author: Jeremy Wu
Version: 3.2.0
Flow.Launcher.Core.ExternalPlugins.FlowPluginException: Method not found: 'Boolean Flow.Launcher.Plugin.SharedCommands.FilesFolders.IsZipFilePath(System.String, Boolean)'.
 ---> System.MissingMethodException: Method not found: 'Boolean Flow.Launcher.Plugin.SharedCommands.FilesFolders.IsZipFilePath(System.String, Boolean)'.
   at Flow.Launcher.Plugin.PluginsManager.PluginsManager.RequestInstallOrUpdate(String search, CancellationToken token, Boolean usePrimaryUrlOnly)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Flow.Launcher.Plugin.PluginsManager.PluginsManager.RequestInstallOrUpdate(String search, CancellationToken token, Boolean usePrimaryUrlOnly)
   at Flow.Launcher.Plugin.PluginsManager.Main.QueryAsync(Query query, CancellationToken token) in D:\a\Flow.Launcher\Flow.Launcher\Plugins\Flow.Launcher.Plugin.PluginsManager\Main.cs:line 51
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 196
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 34
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 195
   --- End of inner exception stack trace ---
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_1.<QueryForPluginAsync>b__1(ActionContext _) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 224
   at Flow.Launcher.Plugin.Result.ExecuteAsync(ActionContext context) in /_/Flow.Launcher.Plugin/Result.cs:line 246
   at Flow.Launcher.ViewModel.MainViewModel.OpenResultAsync(String index) in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 307
   at CommunityToolkit.Mvvm.Input.AsyncRelayCommand.AwaitAndThrowIfFailed(Task executionTask)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

Also I attach a log file.

### To Reproduce

1. See Description section.

### Flow Launcher Version

1.18

### Windows Build Number

22631.3958

### Error Log

[2024-08-14.txt](https://github.com/user-attachments/files/16608059/2024-08-14.txt)

**Comment by deefrawley** [comment]
Are you using the regular or portable version of Flow?

**Comment by Rhynorater** [comment]
I'm also having this problem.

**Comment by cccxp** [comment]
Got same issue on Flow Launcher 1.18.0 portable version. 
`pm update Explorer` will break Explorer Plugin, and `pm update Plugins Manager` will break itself.

- Plugins Manager: 3.1.1 to 3.2.0 
- Explorer: 3.1.6 to 3.2.0

**Comment by royalishere** [comment]
I'm having the same issue on desktop version. Explorer plugin 3.2.0

**Comment by rahulnoir** [comment]
How to downgrade the plugin???

**Comment by Drigva** [comment]
Same issue:

Flow Launcher: 1.18.0 portable version.
Plugins Manager: 3.2.0
Explorer: 3.2.0

**Comment by Drigva** [comment]
> How to downgrade the plugin???

For anyone having the same issue, you use the version 1.17.2 on "...\FlowLauncher\app-1.17.2\Flow.Launcher.exe"

**Comment by nanoraptor** [comment]
For those who want to downgrade to the previous version. Download Explorer v3.1.6 zip from [here](https://github.com/Flow-Launcher/Flow.Launcher.Plugin.Explorer/releases). 

1. Close Flow Launcher from the task manager (All procesess should be terminated)

2. Go to `C:\Users\{User_name}\AppData\Local\FlowLauncher\app-1.18.0\Plugins` Delete `Explorer-3.2.0` and create a new folder `Explorer-3.1.6`
 
3. Transfer/Extract all the zip file contents to the newly created folder `Explorer-3.1.6`

Now start the flow launcher again from the start menu.

**Comment by taooceros** [comment]
You can install with `pm install <url of zip>` directly.

**Comment by nanoraptor** [comment]
> You can install with `pm install <url of zip>` directly.

The manager itself seems to be broken

**Comment by jjw24** [comment]
My apologies everyone, I bumped the Explorer plugin in the dev branch as part of the new release and it appears you updated to the latest Explorer version before I was able to finalise the 1.19.0 release. The updated Explorer plugin actually depends on parts of the Plugin API that is not released yet and that's why there were errors.

To resolve this issue, just upgrade flow to 1.19.0. Future default plugin releases will also follow the timeline of flow's new releases, so we won't get this problem again.
